### ISSUE TITLE
[keycloak] External Database Password missing secret name

### ISSUE THREAD

**Issue by danvy** [bitnami/charts#9421]
### Name and Version

bitnami/keycloak 7.1.2

### What steps will reproduce the bug?

You just have to generate the chart template (`helm template keycloak bitnami/keycloak`) with an external database so `postgresql.enabled=false` and `externalDatabase.password=xxxxxxxxxxxx` (+host, etc.)

### Are you using any custom parameters or values?
```
postgresql.enabled=false
externalDatabase.password=xxxxxxxxxxxx
```
### What is the expected behavior?

The chart should reference the dynamically created secret

### What do you see instead?

This is an extract of the generated statefulset. Notice the empty secretKeyRef name.
```yaml
            - name: KEYCLOAK_ADMIN_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: auth-keycloak
                  key: admin-password
            - name: KEYCLOAK_MANAGEMENT_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: auth-keycloak
                  key: management-password
            - name: KEYCLOAK_DATABASE_PASSWORD
              valueFrom:
                secretKeyRef:
                  **name:** 
                  key: password
```

**Comment by carrodher** [comment]
I'm not able to reproduce the issue:
```console
helm template keycloak bitnami/keycloak --set postgresql.enabled=false,externalDatabase.password=xxxxxxxxxxxx,externalDatabase.host=foo | grep "KEYCLOAK_DATABASE_PASSWORD" -A 4
            - name: KEYCLOAK_DATABASE_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: keycloak
                  key: password
```
This is expected since this parameter is rendered following [this logic](https://github.com/bitnami/charts/blob/0b98d4ce79002f3e17a9b524b4e9fbddf6cc378d/bitnami/keycloak/templates/statefulset.yaml#L222):
```yaml
            - name: KEYCLOAK_DATABASE_PASSWORD
              valueFrom:
                secretKeyRef:
                {{- if .Values.auth.existingSecretPerPassword }}
                  name: {{ tpl (include "common.secrets.name" (dict "existingSecret" .Values.auth.existingSecretPerPassword.databasePassword "context" $)) $ }}
                  key: {{ include "common.secrets.key" (dict "existingSecret" .Values.auth.existingSecretPerPassword "key" "databasePassword") }}
                {{- else }}
                  name: {{ include "keycloak.databaseSecretName" . }}
                  key: {{ include "keycloak.databaseSecretKey" . }}
                {{- end }}
```

**Comment by xAlertax** [comment]
I have same problem 
helm.sh/chart=keycloak-7.1.1
Image:       docker.io/bitnami/keycloak:16.1.1-debian-10-r33

`  Warning  FailedCreate      6s (x190 over 2d1h)  statefulset-controller  create Pod keycloak-test-2 in StatefulSet keycloak-test failed error: Pod "keycloak-test-2" is invalid: spec.containers[0].env[4].valueFrom.secretKeyRef.name: Invalid value: "": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')`

**Comment by migruiz4** [comment]
Hi, this issue has been already fixed in version 7.1.3 of the `bitnami/keycloak` chart with this PR https://github.com/bitnami/charts/pull/9383

Due to some issues with our release process, it wasn't included in version 7.1.2 but 7.1.3.
